pylint/message/message_definition.py

```diff
@@ -70,11 +70,11 @@ def __repr__(self) -> str:
     def __str__(self) -> str:
         return f"{repr(self)}:\n{self.msg} {self.description}"
 
-    def may_be_emitted(self) -> bool:
-        """Return True if message may be emitted using the current interpreter."""
-        if self.minversion is not None and self.minversion > sys.version_info:
+    def may_be_emitted(self, py_version: tuple = sys.version_info) -> bool:
```

Comment:
Please refer to the last paragraph of the PR description. How would I pass in py_version in the case of message_definition_store.py:find_emittable_messages in code `if message.may_be_emitted():` ? This is why I kept version_info as default, to remain consistent with previous code.

Reply:
> Isn't `sys.version_info`a nice enough default ? Basically it's the default value we had before implicitly.

That's my belief too :) It doesn't break any code. This PR isn't meant to magically fix everything, it's incremental.

Reply:
Yeah, that's also my concern: is this considered public API.

Imo, this is actually making the default implicit. The default of the option is explicit and this is what gets propagated to the function call. By adding the default here we add another source of a potential default: compared to the situation pre-PR there is a new source that could impact the `py-version` behaviour. In a way adding the default actually complicates the situation. I'm not sure that is an actual concern but I did want to raise a point about it as I think this can be easily overlooked.

Reply:
One solution to this could be to deprecate the default value like this:
```python
def may_be_emitted(self, py_version: tuple | None = None) -> bool:
    if py_version is None:
        py_version =  sys.version_info
        warning.warn("Deprecation yadayada 'py_version' will be a required parameter in 3.0 yada yada...")
```
Or a more global refactor where we handle py_version in the message store and do not instantiate message definitions that will never be emitted.

Reply:
I think that would be good indeed!

Reply:
How about we take a decision about it in another issue and merge this as is considering there is no change in behavior ? The fast "fix" is deprecation but I'm not sure that forcing every checker to give the ``py_version`` from their config should be the final solution in 3.0, so I would favor the refactor long term. Let's open an issue to discuss this ?

Reply:
Let's do the deprecation here. `may_be_emitted` shouldn't really be used anyway: checkers should use `is_message_enabled` which itself doesn't even call `may_be_emitted`. So this is really only a method that we use internally but which isn't prefixed with `_` correctly. It's not too much effort to apply your suggestion (which I think is quite elegant).

Reply:
[b4f9022](https://github.com/PyCQA/pylint/pull/7580/commits/b4f9022e3cf2ee9580ca53f90f572d1a5729d838) :)